File: 1331-rank-transform-of-an-array/1331-rank-transform-of-an-array.cpp
class Solution {
public:
    vector<int> arrayRankTransform(vector<int>& arr) {
        map<int, int> mp;
        for (auto it : arr) {
            mp[it]++;
        }
        vector<int> res;
        int i = 1;
        for (auto& it : mp) {
            it.second = i;
            i++;
        }
        for (auto it : arr) {
            res.push_back(mp[it]);
        }
        return res;
    }
};

File: 1331-rank-transform-of-an-array/1331-rank-transform-of-an-array.java
class Solution {
    public int[] arrayRankTransform(int[] arr) {
        int n = arr.length;
        int[] result = arr.clone();
        Arrays.sort(result);
        Map<Integer, Integer> map = new HashMap<>();
        int rank = 1;
        for (int i = 0; i < n; i++) {
            if (!map.containsKey(result[i])) {
                map.put(result[i], rank);
                rank += 1;
            }
        }
        for (int i = 0; i < n; i++) {
            result[i] = map.get(arr[i]);
        }
        return result;
    }
}